Packet++/src/S7CommLayer.cpp

```diff
@@ -0,0 +1,117 @@
+#include "EndianPortable.h"
+
+#include "S7CommLayer.h"
+#include "TcpLayer.h"
+#include <iostream>
+
+#include <cstring>
+#include <sstream>
+
+namespace pcpp
+{
+
+	S7CommLayer::S7CommLayer(uint8_t msgType, uint16_t pduRef, uint16_t paramLength, uint16_t dataLength,
+							 uint8_t errorClass, uint8_t errorCode)
+	{
+		size_t headerLen;
+		if (msgType == 0x03)
+		{
+			headerLen = sizeof(s7comm_ack_data_hdr);
+			m_DataLen = headerLen;
```

Comment:
The data len is not correct I think, because the actual length should be: `sizeof(s7comm_ack_data_hdr) + paramLength + dataLength`. We should also set `m_Data` to this size

Reply:
@wivien19 you didn't fix this one